### PR TITLE
testutils: resourcebuilder: machine add providerID

### PR DESCRIPTION
**Pull request by damdo**
Add the ability to specify the providerID for a Machine.

### CODE REVIEW

**Review by elmiko — COMMENTED** [review]
/lgtm

but let's make sure we didn't break any of the tests.

File: testutils/resourcebuilder/machine/v1beta1/machine.go [hunk 1]
```diff
@@ -140,6 +145,12 @@ func (m MachineBuilder) WithProviderSpecBuilder(builder resourcebuilder.RawExten
 	return m
 }
 
+// WithProviderID sets the providerID field for the machine builder.
+func (m MachineBuilder) WithProviderID(providerID string) MachineBuilder {
```
Comment (JoelSpeed) [inline testutils/resourcebuilder/machine/v1beta1/machine.go:149]: Nit, ProviderID is alphabetically ahead of ProviderSpec. I believe everything else here is alphabetically sorted so nice to keep that

Comment (damdo) [reply testutils/resourcebuilder/machine/v1beta1/machine.go:143]: Ah yeah sure :) updated.

**Review by JoelSpeed — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**